### PR TITLE
Implemented copy code button feature

### PR DESCRIPTION
**Pull request by george-gca**
Implemented support for copy code button in code blocks (#1262), also updated blog post about code to reflect it.

Here's how it looks like in light mode:

image

image

And in dark mode:

image

image

It does only appears when displaying code blocks using the markdown syntax, like:
````markdown
```c++
code code code
```
````

### CODE REVIEW

**Comment by george-gca** [comment]
Just now realized that my vscode formatted the css file differently. If you want to, I can undo the formatting.

File: _layouts/post.html [hunk 1]
```diff
@@ -57,3 +57,5 @@ <h1 class="post-title">{{ page.title }}</h1>
   {%- endif -%}
 
 </div>
+
+<script defer src="{{ '/assets/js/copy_code.js' | relative_url }}" type="text/javascript"></script>
```
Comment (alshedivat) [inline _layouts/post.html:61]: this line should go to `_includes/scripts/misc.html`, which will then be included in the default layout: https://github.com/alshedivat/al-folio/blob/master/_layouts/default.html#L30

File: _posts/2015-07-15-code.md [hunk 2]
```diff
@@ -39,3 +69,5 @@ int main(int argc, char const \*argv[])
 }
 
 {% endhighlight %}
+
+Note that when using the liquid tag, the `copy code` button is not displayed.
```
Comment (alshedivat) [inline _posts/2015-07-15-code.md:73]: do you know why this is the case? in principle, is it possible to make `copy_code.js` work with both types of code blocks?

File: _sass/_base.scss
[the remark this answers is not in the document]
Comment (george-gca) [reply _sass/_base.scss:1]: Actually I use the VSCode default formatter, I just changed to format before saving the file. But adding a linter/formatter would be a nice idea to keep the code tidy. The VSCode formatter even change the current formatting of the README file every time I save, so I always undo some of them before commiting.

File: _posts/2015-07-15-code.md (hunk 2 above)
Comment (george-gca) [reply _posts/2015-07-15-code.md:73]: I am investigating this now, but apparently this is a [known issue](https://github.com/jekyll/jekyll/issues/6448) in Jekyll.

Under the hood though, I found out that this is the difference (when using the c++ example):
- Using markdown code block
```html
<pre class="highlight"><code>
```
- Using jekyll highlight
```html
<pre><code class="language-c--" data-lang="c++">
```

Maybe just removing the filter with `highlight` class works? Will try it now.

**Comment by george-gca** [comment]
Made requested changes.
Now I realized that there is an extra space after code when using line numbers (don't know why this happens, but it always did).

image

**Review by alshedivat — APPROVED** [review]
lgtm and thanks again for the contribution! the extra space at the end of numbered code block was due to a spurious margin.

**Pull request merged**